spec.html

```diff
@@ -28440,6 +28440,209 @@ <h1>
+                <th>Value Type</th>
+                <th>Meaning</th>
+              </tr>
+            </thead>
+            <tr>
+              <td>[[ExportNames]]</td>
+              <td>a List of Strings</td>
+              <td>The names of the exports of the module. This list does not contain duplicates.</td>
+            </tr>
+            <tr>
+              <td>[[EvaluationSteps]]</td>
+              <td>an Abstract Closure</td>
+              <td>The initialization logic to perform upon evaluation of the module, taking the Synthetic Module Record as its sole argument. These will usually set up the exported values, by using SetSyntheticModuleExport. It must not modify [[ExportNames]]. It may return an abrupt completion.</td>
+            </tr>
+          </table>
+        </emu-table>
+
+        <emu-clause id="sec-createsyntheticmodule" type="abstract operation">
+          <h1>
+            CreateSyntheticModule (
+              _exportNames_: a List of Strings without duplicates,
+              _evaluationSteps_: an Abstract Closure,
+              _realm_: a Realm Record,
+            ): a Synthetic Module Record
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd></dd>
+          </dl>
+
+          <emu-alg>
+            1. Return the Synthetic Module Record { [[Realm]]: _realm_, [[Environment]]: ~empty~, [[Namespace]]: ~empty~, [[HostDefined]]: *undefined*, [[ExportNames]]: _exportNames_, [[EvaluationSteps]]: _evaluationSteps_ }.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-create-default-export-synthetic-module" type="abstract operation">
+          <h1>
+            CreateDefaultExportSyntheticModule (
+              _defaultExport_: an ECMAScript Language value,
+            ): a Synthetic Module Record
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>It creates a Synthetic Module Record whose default export is _defaultExport_.</dd>
+          </dl>
+          <emu-alg>
+            1. Let _realm_ be the current Realm Record.
+            1. Let _setDefaultExport_ be a new Abstract Closure with parameters (_module_) that captures _defaultExport_ and performs the following steps when called:
+              1. Perform SetSyntheticModuleExport(_module_, *"default"*, _defaultExport_).
+            1. Return CreateSyntheticModule(« *"default"* », _setDefaultExport_, _realm_).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-parse-json-module" type="abstract operation">
+          <h1>
+            ParseJSONModule (
+              _source_: a String,
+            ): either a normal completion containing a Synthetic Module Record, or a throw completion
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd></dd>
+          </dl>
+
+          <emu-alg>
+            1. Let _json_ be ? Call(%JSON.parse%, *undefined*, « _source_ »).
```

Comment:
Should steps 2-9 of [JSON.parse](https://tc39.es/ecma262/#sec-json.parse) be extracted to a new AO that we can call here, or is this ok?

Reply:
I'm fine with either way.

Reply:
I mildly prefer the AO (after https://github.com/tc39/ecma262/pull/3394 lands), but that refactoring can happen after this PR.